WordPress/Classes/System/WordPressAppDelegate.m

```diff
@@ -434,6 +436,35 @@ - (void)application:(UIApplication *)application handleActionWithIdentifier:(NSS
 #pragma mark - OpenURL helpers
 
 /**
+
+ */
+- (BOOL)handleOpenWithAuthenticationURL:(NSURL *)url
+{
+    DDLogInfo(@"App launched with authentication link");
+    NSParameterAssert([url isKindOfClass:[NSURL class]]);
+    NSDictionary *params = [[url query] dictionaryFromQueryString];
+
+    NSString *token = [params objectForKey:@"token"];
+    NSAssert(token != nil, @"The URL did not have the expected path.");
+
+    // if already logged in, do nothing.
+    AccountService *service = [[AccountService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
+    if (service.defaultWordPressComAccount) {
+        DDLogInfo(@"App opened with authentication link but there is already an existing wpcom account. %@", service.defaultWordPressComAccount);
+        return YES;
+    }
+
+    // Bump stat for opening the app via a magic link.
+    [WPAppAnalytics track:WPAnalyticsStatLoginMagicLinkOpened];
+
+    // Show the signin view controller configured to perform the auth request.
+    [self showSigninScreen:token animated:NO thenEditor:NO];
+
+    return YES;
+}
+
+
+/**
```

Comment:
I wasn't sure which method this note was referring to so I made a best guess.  Let me know if I chose the wrong one.